mlflow/pyfunc/__init__.py

```diff
@@ -397,6 +402,15 @@ def _enforce_mlflow_datatype(name, values: pandas.Series, t: DataType):
             raise MlflowException(
                 "Failed to convert column {0} from type {1} to {2}.".format(name, values.dtype, t)
             )
+    if t == DataType.double and values.dtype == decimal.Decimal:
```

Comment:
I thought in our test the `values.dtype` is `object`, while `type(values[0])` is `decimal.Decimal`. Can you confirm?

Reply:
yeah, verified
<img width="549" alt="image" src="https://user-images.githubusercontent.com/109561646/186981589-5af8aa09-8b0f-48c6-b437-44de44dbb13f.png">

Reply:
Also our unit test and e2e notebook test all passed when object is Decimal

Reply:
Ah, I didn't know `dtype == object` and `dtype == decimal.Decimal` can both be true. LGTM!